semeio/jobs/scripts/auto_correlate_observations.py

```diff
@@ -0,0 +1,75 @@
+import yaml
+
+from ert_data.measured import MeasuredData
+from ert_shared.libres_facade import LibresFacade
+from res.enkf import ErtScript
+
+from semeio.jobs import auto_correlate_observations
+from semeio.jobs.scripts.correlated_observations_scaling import (
+    CorrelatedObservationsScalingJob,
+)
+from semeio.jobs.correlated_observations_scaling.exceptions import EmptyDatasetException
+
+
+class AutoCorrelateObservationsJob(ErtScript):
+    def run(self, *args):
+        config_record = _fetch_config_record(args)
+        measured_record = _load_measured_record(self.ert())
+        scaling_configs = auto_correlate_observations.run(
+            **{
+                "auto_correlation_config": config_record,
+                "measured_data": measured_record,
+            }
+        )
+
+        # The execution of COS should be moved into
+        # auto_correlate_observations.run when COS no longer depend on self.ert
+        # to run.
+        scaling_params = _fetch_scaling_parameters(config_record, measured_record)
+        for scaling_config in scaling_configs:
+            scaling_config["CALCULATE_KEYS"].update(scaling_params)
+
+        try:
+            CorrelatedObservationsScalingJob(self.ert()).run(scaling_configs)
+        except EmptyDatasetException:
+            pass
+
+
+def _fetch_scaling_parameters(config_record, measured_data):
+    config = auto_correlate_observations.assemble_auto_correlate_config(
+        config_record, measured_data,
+    )
+    if not config.valid:
+        # The config is loaded by auto_correlate_observations.run first. The
+        # second time should never fail!
+        raise ValueError("AutoCorrelateConfig not valid on second load")
+
+    scale_conf = config.snapshot.scaling
+    return {
+        "threshold": scale_conf.threshold,
```

Comment:
fetch strings from config namespace

Reply:
This string actually comes from the COS job and that schema consists of magic strings. So there is no possibility of fetching this without doing some work in the schema. See https://github.com/equinor/semeio/blob/e6a958997703d8bcb2cef3c57de05f8034c87681/semeio/jobs/correlated_observations_scaling/job_config.py#L120

Persist it in an issue instead?

Reply:
👍